hongyeollee/app.js

```diff
@@ -33,21 +33,136 @@ app.get("/ping", (req, res) => {
   res.json({ message: "pong" });
 });
 
-//create user
+//과제 2.create user
 app.post("/signup", async (req, res) => {
-  const { name, password, age, email } = req.body;
+  const { name, password, profileImage, email } = req.body;
 
   await mysqlDatabase.query(
     `INSERT INTO users(
           name,
           password,
-          age,
+          profile_image,
           email
         )VALUES(?, ?, ?, ?);
         `,
-    [name, password, age, email]
+    [name, password, profileImage, email]
   );
-  res.status(201).json({ message: "userCreated ✅" });
+  res.status(201).json({ message: "userCreated 👫" });
+});
+
+//과제 3. create post
+app.post("/post", async (req, res) => {
+  const { title, content, imageUrl } = req.body;
+
+  await mysqlDatabase.query(
+    `INSERT INTO posts(
+          title,
+          content,
+          image_url
+        )VALUES(?, ?, ?);
+        `,
+    [title, content, imageUrl]
+  );
+  res.status(201).json({ message: "postCreated 📝" });
+});
+
+//과제 4. search and get posts
+app.get("/getpost", async (req, res) => {
+  await mysqlDatabase.query(
+    `SELECT
+        u.id AS userId,
+        u.profile_image AS userProfileImage,
+        p.id AS postingId,
+        p.image_url AS postingImageUrl,
+        p.content AS postingContent
+    FROM users u
+    INNER JOIN posts p
+    ON u.id=p.id`,
+    (err, rows) => {
+      res.status(200).json({ data: rows });
+    }
+  );
+});
+
+//과제 5. get post of targeting user
+app.get("/postofuser", async (req, res) => {
```

Comment:
특정 유저의 게시물을 조회하는 기능으로 보입니다. 현재 엔드포인트에서는 어떻게 user id값을 가져오는걸까요?
request의 param으로 user의 id를 가져와 보세요! 
```suggestion
app.get("/posts/user/:userId", async (req, res) => {
```
해당 엔드포인트는 게시글, 유저, 유저ID 값이 모두 보여지고 있습니다. 그렇기에
이는 특정한 유저의 게시글을 가져오는 API임을 유추할 수 있습니다.